apisix/core/config_etcd.lua

```diff
@@ -181,10 +181,6 @@ local function sync_data(self)
             return false, err
         end
 
-        if not dir_res.dir then
-            return false, self.key .. " is not a dir"
-        end
-
```

Comment:
Currently I still set `dir = true` in [etcd protocol conversion](https://github.com/apache/apisix/blob/9ac02f66b14fa3c1b5a8f9341a82b664ceda17fb/apisix/core/etcd.lua#L61) for consistency between protocols. But I noticed there is only one test case ([global.t TEST 3](https://github.com/apache/apisix/blob/9ac02f66b14fa3c1b5a8f9341a82b664ceda17fb/t/admin/global-rules.t#L125-L127)) needs this parameter, so maybe we could remove it